lib/Backend/Encoder.cpp

```diff
@@ -337,6 +335,8 @@ Encoder::Encode()
         XDataAllocator::Register(&alloc->allocation->xdata, m_func->GetJITOutput()->GetCodeAddress(), m_func->GetJITOutput()->GetCodeSize());
         m_func->GetInProcJITEntryPointInfo()->SetXDataInfo(&alloc->allocation->xdata);
     }
+
+    m_func->GetJITOutput()->SetCodeAddress(m_func->GetJITOutput()->GetCodeAddress() | 0x1); // Set thumb mode
```

Comment:
Sorry, my initial description was not quite right. The issue is not with the xdata itself, but with registration of the xdata with the OS. The registration API doesn't want the thumb encoded address.